logstash-core/lib/logstash/agent.rb

```diff
@@ -10,80 +10,125 @@
 LogStash::Environment.load_locale!
 
 class LogStash::Agent
-  attr_reader :logger
+  attr_reader :logger, :state
 
-  def initialize(logger)
-    @logger = logger
+  def initialize(params)
+    @logger = params[:logger]
     @pipelines = {}
+    @pipeline_threads = {}
+    @state = clean_state
```

Comment:
Some documentation can set the right expectations, as this is more of an internal representation of the agent state, and not the current status or "lifetime phase".

Here `state` means the same as a node state or cluster state in elasticsearch, which is a document string or hash that represents the internal state.
This can also be interpreted as the State in erlang's OTP framework: any kind of data structure that represents state.

My problem with calling this current_config is that another agent can have much more metadata about its state than just the config itself (versioning, timestamps, last known error, etc)

what do you think of just adding a comment setting this expectation?